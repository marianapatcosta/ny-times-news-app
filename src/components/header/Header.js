import { defineComponent } from "vue";

export default defineComponent({
  name: "Header",
  props: {
    title: {
      type: String,
      required: true,
    },
  },
});
