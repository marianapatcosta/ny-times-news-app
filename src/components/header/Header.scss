header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 10;
  width: 100%;
  height: 4rem;
  padding: 1.2rem 1rem;
  box-shadow: 0 0.125rem 0.5rem rgba($color-black, 0.7);
  background-color: $color-secondary;
  color: $color-primary;

  h1 {
    font-size: 180%;
    font-style: italic;
    user-select: none;
  }
}

@media only screen and (min-width: $sm) {
  header {
    padding: 1.2rem 3.5rem;
  }
}
