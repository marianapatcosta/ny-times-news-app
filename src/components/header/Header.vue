<template>
  <header>
    <h1>{{ title }}</h1>
  </header>
</template>

<script src="./Header.js"></script>

<style scoped lang="scss" src="./Header.scss"></style>
