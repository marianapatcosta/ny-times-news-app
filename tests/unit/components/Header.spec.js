import { shallowMount } from "@vue/test-utils";
import Header from "@/components/header/Header.vue";

describe("Header.vue", () => {
  const title = "Header Title";
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Header, {
      props: { title },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });
  it("should render without errors and match snapshot renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render props.title when passed", () => {
    expect(wrapper.text()).toMatch(title);
  });
});
