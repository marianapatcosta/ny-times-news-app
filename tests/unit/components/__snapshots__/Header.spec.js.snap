// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue should render without errors and match snapshot renders correctly 1`] = `
<header>
  <h1>
    Header Title
  </h1>
</header>
`;
